tests/hybrid/test_detailed_pv.py

```diff
@@ -97,23 +146,98 @@ def test_load_data(site):
     pv_plant.total_installed_cost = cost_model.calculate_total_costs(bos_data)
 
-    pv_plant.simulate_financials(5000, 1)
+    pv_plant.simulate_financials(5000, 1)       # TODO: implement a simple NPV calculator as an example
 
     pv_plant.export_financials()
     
     new_pv_design_vector = {
         'system_capacity_kw': 5000,
-        'layout_params': DetailedPVParameters(x_position=0.5,
-                                             y_position=0.5,
-                                             aspect_power=0,
-                                             s_buffer=0.1,
-                                             x_buffer=0.1,
-                                             gcr=0.7,
-                                             azimuth=180,
-                                             tilt_tracker_angle=45,
-                                             string_voltage_ratio=0.4,
-                                             dc_ac_ratio=1.3
-                                             )}
-
-    pv_plant._layout.set_layout_params(new_pv_design_vector['system_capacity_kw'], new_pv_design_vector['layout_params'])
-                                            
\ No newline at end of file
+        'layout_params': DetailedPVParameters(
+            x_position=0.5,
+            y_position=0.5,
+            aspect_power=0,
+            s_buffer=0.1,
+            x_buffer=0.1,
+            gcr=0.7,
+            azimuth=180,
+            tilt_tracker_angle=45,
+            string_voltage_ratio=0.4,
+            dc_ac_ratio=1.3)
+        }
+
+    pv_plant._layout.set_layout_params(
+        new_pv_design_vector['system_capacity_kw'],
+        new_pv_design_vector['layout_params']
+        )
+
+    # Simulate again to ensure layout changes propagate
+    pv_plant.simulate_power(1, False)
+
+    assert pv_plant._system_model.Outputs.annual_energy == approx(9445701.25731733, 1e-1)
+    assert pv_plant._system_model.Outputs.capacity_factor == approx(21.57259825712787, 1e-1)
+
+
+def test_hybrid_pv_plants(site):
+    pvsamv1_defaults_file = Path(__file__).absolute().parent / "pvsamv1_basic_params.json"
+    with open(pvsamv1_defaults_file, 'r') as f:
+        tech_config = json.load(f)
+
+    design_vec = DetailedPVParameters(
+        x_position=0.25,
+        y_position=0.5,
+        aspect_power=0,
+        s_buffer=0.1,
+        x_buffer=0.1,
+        gcr=0.3,
+        azimuth=180,
+        tilt_tracker_angle=0,
+        string_voltage_ratio=0.5,
+        dc_ac_ratio=1.2)
+
+    pv_config = {
+        'tech_config': tech_config,
+        'layout_params': design_vec,
+        'layout_config': default_layout_config,
+        'fin_config': default_fin_config,
+        'pan_file': "",
+        'ond_file': "",
+    }
+
+    annual_energy_expected = 108239401.3
+
+    # Test standalone DetailedPVPlant 
+    pv_plant = DetailedPVPlant(site=site, pv_config=pv_config)
+    pv_plant.simulate_power(1, False)
```

Comment:
Layout fxn is needed here too. Normally the layout fxn is called from within HybridSimulation, but here we have to do it manually

Reply:
Fixed by the aforementioned call in processed_assign().